File: .sdk-repos/orchestration-cluster-api-rust/examples/secret.rs
//! Compilable usage examples for secret operations.
//!
//! Region-tagged snippets are surfaced in the Camunda REST API documentation via
//! `examples/operation-map.json`, and are type-checked by `cargo build --examples`.
#![allow(dead_code, unused_variables, unused_imports)]

use camunda_orchestration_sdk::apis::secret_api::ListSecretsParams;
use camunda_orchestration_sdk::apis::secret_api::ResolveSecretsParams;
use camunda_orchestration_sdk::models::SecretResolveRequest;
use camunda_orchestration_sdk::CamundaClient;

// region ResolveSecrets
async fn resolve_secrets() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .resolve_secrets(ResolveSecretsParams {
            secret_resolve_request: SecretResolveRequest {
                references: vec!["camunda.secrets.my-secret".to_string()],
            },
        })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion ResolveSecrets

// region ListSecrets
async fn list_secrets() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .list_secrets(ListSecretsParams { body: None })
        .await?;
    println!("{result:#?}");

    Ok(())
}
// endregion ListSecrets

fn main() {
    // Examples above are compiled, not executed.
}

File: .sdk-repos/orchestration-cluster-api-php/examples/secret.php
<?php

/**
 * Compilable usage examples for secret resolution operations.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;

// region ResolveSecrets
/**
 * Resolve secrets.
 */
function resolve_secrets(CamundaClient $client, \Camunda\Orchestration\Api\Model\SecretResolveRequest $secretResolveRequest): void
{
    $client->resolveSecrets($secretResolveRequest);
}
// endregion ResolveSecrets

// region ListSecrets
/**
 * List secret references.
 *
 * @param array<string, mixed>|null $body
 */
function list_secrets(CamundaClient $client, ?array $body = null): void
{
    $client->listSecrets($body);
}
// endregion ListSecrets
